src/GHCMod.hs

```diff
@@ -333,7 +342,14 @@ main = handler $ do
 
 progMain :: (Options,[String]) -> IO ()
 progMain (globalOptions,cmdArgs) = do
-    (res,_) <- runGhcModT globalOptions $ ghcCommands cmdArgs
+    let
```

Comment:
I see no reason to. If I just move part of this code to runGhcModT invocation, it would simplify things a fair bit.